parlai/agents/image_seq2seq/image_seq2seq.py

```diff
@@ -0,0 +1,349 @@
+
+    def load_state_dict(self, state_dict: Dict[str, torch.Tensor]):
+        """
+        Override for custom loading.
+
+        Three reasons:
+            1. When using an init model without an image encoder
+            2. When using an init model with only an encoder provided
+                In this case, we may need to add the START token to the state_dict
+            3. When using an init model without image tokens in the embeddings
+        """
+        # Case 1 -> No Image Encoder
+        if 'encoder.image_encoder.0.weight' not in state_dict:
+            for k, v in self.model.encoder.image_encoder.state_dict().items():
+                state_dict[f'encoder.image_encoder.{k}'] = v
+
+        # Case 2 -> Only an Encoder provided
+        if not (any('decoder' in state_key for state_key in state_dict)):
+            for k, v in self.model.decoder.state_dict().items():
+                state_dict[f'decoder.{k}'] = v
+            state_dict['decoder.embeddings.weight'] = state_dict['embeddings.weight']
+            if 'START' not in state_dict:
+                state_dict['START'] = self.model.START
+
+        cnt = 0
+        if self.opt['init_model'] is not None:
+            try:
+                self.model.load_state_dict(state_dict)
+                return
+            except Exception as e:
+                # Case 3 --> Check for Embedding Diffs
+                print('exception in loading state dict: {}'.format(e))
+                # make sure dims match up
+                embs = state_dict['embeddings.weight']
+                enc_embs = state_dict['encoder.embeddings.weight']
+                dec_embs = state_dict['decoder.embeddings.weight']
+                num_embs = min(self.model.embeddings.weight.shape[0], embs.shape[0])
+                num_enc_embs = min(
+                    self.model.encoder.embeddings.weight.shape[0], enc_embs.shape[0]
+                )
+                num_dec_embs = min(
+                    self.model.decoder.embeddings.weight.shape[0], dec_embs.shape[0]
+                )
+                for i in range(num_embs):
+                    self.model.embeddings.weight.data[i] = embs[i]
+                    cnt += 1
+                for i in range(num_enc_embs):
+                    self.model.encoder.embeddings.weight.data[i] = enc_embs[i]
+                for i in range(num_dec_embs):
+                    self.model.decoder.embeddings.weight.data[i] = dec_embs[i]
+                state_dict['embeddings.weight'] = self.model.embeddings.weight
+                state_dict[
+                    'encoder.embeddings.weight'
+                ] = self.model.encoder.embeddings.weight
+                state_dict[
+                    'decoder.embeddings.weight'
+                ] = self.model.decoder.embeddings.weight
+
+                print(
+                    'Initialized embeddings for {} tokens ({}%).'
+                    ''.format(cnt, round(cnt * 100 / len(self.dict), 1))
```

Comment:
Consider `f'Initialized embeddings for {cnt} tokens ({cnt/len(self.dict).1%})'`

Reply:
(I personally don't like expressions in f-strings, I prefer to rip out the expression into a temp variable, but the `{var:.1%}` is what I want to highlight)